### PR TITLE
More documentation updates

### PR DESCRIPTION
**Pull request by cfergeau**
This removes the obsolete mentions of 'libvirtd' group on ubuntu/debian, add minimum version requirements for misc distros, and try to document better what our goal is.

### CODE REVIEW

File: docs/source/getting-started/content/con_understanding-codeready-containers.adoc [hunk 1]
```diff
@@ -12,4 +14,5 @@
 * It uses a single node which behaves both as a master and as a worker node.
 * The `machine-config`, `marketplace` and `monitoring` Operators are disabled by default.
 * These disabled Operators will cause the corresponding parts of the web console to be non functional.
+* For the same reason, there is currently no upgrade path to newer OpenShift versions.
```
Comment (gbraad) [inline docs/source/getting-started/content/con_understanding-codeready-containers.adoc]: in general, we do not have an upgrade strategy (yet). due to the expiring certificates we have to make people aware of the ephemeral nature; else we also have to take care of moving images out and in to the new updated image?

File: docs/source/getting-started/content/assembly_troubleshooting.adoc [hunk 2]
```diff
@@ -1,4 +1,10 @@
 [id="troubleshooting-codeready-containers_{context}"]
 = Troubleshooting {rh-prod}
 
+[NOTE]
+====
+{rh-prod}'s goal is to deliver an OpenShift environment for development and testing purposes.
+Issues occurring during installation or usage of specific OpenShift applications are outside of {prod}'s realm and should be reported to the relevant project.
```
Comment (gbraad) [inline docs/source/getting-started/content/assembly_troubleshooting.adoc]: we might need to be more specific and point to the actual locations to report issues for OpenShift itself, etc.

File: docs/source/getting-started/content/con_understanding-codeready-containers.adoc [hunk 3]
```diff
@@ -2,6 +2,8 @@
 = Understanding {prod}
 
 {rh-prod} brings a minimal OpenShift 4.0 or newer cluster to your local computer.
+This cluster provides a minimal environment for development and testing purposes.
+It may or may not be a good fit for other usecases, and is not a replacement for the https://cloud.redhat.com/openshift/install/[full-fledged OpenShift installer].
```
Comment (gbraad) [inline docs/source/getting-started/content/con_understanding-codeready-containers.adoc]: this statement is right, but sounds harsh... I am not sure how to rephrase; tried to reverse but that doesn't sound good about what this is about, but I am kinda inclined to mention a "headless, multi-developer setup" / "team-based"

File: pkg/crc/preflight/preflight_checks_linux.go [hunk 4]
```diff
@@ -170,7 +170,6 @@ func checkUserPartOfLibvirtGroup() (bool, error) {
 
 func fixUserPartOfLibvirtGroup() (bool, error) {
 	logging.Debug("Adding current user to the libvirt group")
-	// Add user to libvirt/libvirtd group based on distro
```
Comment (gbraad) [inline pkg/crc/preflight/preflight_checks_linux.go]: This only applies to later releases of Ubuntu/debian. We need to confirm when this happened.

Comment (cfergeau) [reply pkg/crc/preflight/preflight_checks_linux.go]: This happened in 16.10, I thought I added that quote to the commit log, but apparently not. The Debian developer I asked  could not find any traces of a 'libvirtd' group in the package history, so let's assume they always used 'libvirt'

File: docs/source/getting-started/content/assembly_troubleshooting.adoc (hunk 2 above)
Comment (cfergeau) [reply docs/source/getting-started/content/assembly_troubleshooting.adoc]: Can add an OpenShift pointer, but OpenShift applications will be spread among a number of locations, so hard to give any pointers for that.

Comment (praveenkumar) [reply docs/source/getting-started/content/assembly_troubleshooting.adoc]: Yes, there are lot of openshift projects/components, it is really hard to put any specific pointer.

File: docs/source/getting-started/content/con_understanding-codeready-containers.adoc (hunk 3 above)
Comment (cfergeau) [reply docs/source/getting-started/content/con_understanding-codeready-containers.adoc]: Ah forgot to change that part, I'll update the PR a second time

Comment (cfergeau) [reply docs/source/getting-started/content/con_understanding-codeready-containers.adoc]: I currently have:
> Red Hat CodeReady Containers brings a minimal OpenShift 4.0 or newer cluster to your local computer. This cluster provides a minimal environment for development and testing purposes. It’s mainly targetted at running on developers' desktops. For other use cases, such as headless, multi-developer/team-based setups, …​, use of the full-fledged OpenShift installer is recommended.

File: docs/source/getting-started/content/proc_installing-codeready-containers.adoc [hunk 5]
```diff
@@ -20,6 +20,8 @@ Other applications running on top of OpenShift may require additional resources
 * On {mac}, {prod} requires macOS 10.12 Sierra or newer.
 {prod} does not work on earlier versions of {mac}.
 
+* On Linux, {prod} is only supported on {rhel}/{centos} 7.5 or newer (including 8.x versions) and on the latest two stable {fed} releases.
+"{ubuntu} 18.04 LTS or newer and {debian} 10 or newer are not officially supported and may require manual set up of the host machine.
```
Comment (robin-owen) [inline docs/source/getting-started/content/proc_installing-codeready-containers.adoc]: There's a stray quotation mark at the beginning of the sentence here. ;)

File: docs/source/getting-started/content/assembly_troubleshooting.adoc [hunk 6]
```diff
@@ -1,4 +1,11 @@
 [id="troubleshooting-codeready-containers_{context}"]
 = Troubleshooting {rh-prod}
 
+[NOTE]
+====
+{rh-prod}'s goal is to deliver an OpenShift environment for development and testing purposes.
```
Comment (robin-owen) [inline docs/source/getting-started/content/assembly_troubleshooting.adoc]: Apologies, I missed this before. We should try to avoid CodeReady Containers showing possession of anything, so I'd reword this slightly:

```adoc
The goal of {rh-prod} is to....
```

Beyond that, this PR looks good to me. :+1:
